### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by dkischenko**

### CODE REVIEW

File: internal/user/service_test.go [hunk 1]
```diff
@@ -665,3 +670,81 @@ func TestService_CreateTokenUpdateKeyErr(t *testing.T) {
...
+		}
+	)
+
+	var cfg *config.Config
+
+	configPath := os.Getenv("CONFIG")
+	var once sync.Once
+	once.Do(func() {
+		cfg = config.GetConfig(configPath, &config.Config{})
+	})
```
Comment (Dimitriy14) [inline internal/user/service_test.go:689]: Why do we need once here?

File: internal/user/service_test.go [hunk 2]
```diff
@@ -665,3 +670,81 @@ func TestService_CreateTokenUpdateKeyErr(t *testing.T) {
...
+		ctx = context.Background()
+	)
+
+	var cfg *config.Config
+
+	configPath := os.Getenv("CONFIG")
+	var once sync.Once
+	once.Do(func() {
+		cfg = config.GetConfig(configPath, &config.Config{})
+	})
```
Comment (Dimitriy14) [inline internal/user/service_test.go:726]: the same as above

File: internal/user/service_test.go [hunk 3]
```diff
@@ -665,3 +670,81 @@ func TestService_CreateTokenUpdateKeyErr(t *testing.T) {
...
+		cfg.Storage.Password, cfg.Storage.Database)
+	if err != nil {
+		panic(err)
+	}
+	storage := database.NewStorage(client, l)
+
+	s := user.NewService(l, storage, accessTokenTTL)
+
+	for i := 0; i < b.N; i++ {
+		_, _ = s.FindByUUID(ctx, "e103da7d-5336-422c-abb5-a2d2d26c1786")
```
Comment (Dimitriy14) [inline internal/user/service_test.go:748]: Are you sure that full flow is passing? What if this func is failing with error on the first step?

File: profiling/summary.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Summary for profiling.
+During investigation profile my application I didn't find any big troubles with CPU/Memory. 
+For memory optimization I can use in application library [fastjson](https://github.com/valyala/fastjson) for validation and parsing JSON data instead of `json.Decode` and `go-playground/validator`.
```
Comment (Dimitriy14) [inline profiling/summary.md:3]: Describe cases where you used concurrency and why.
What about logging? Does current logger is able to write logs asynchronously? Which libs you can also use?
Describe why you used standart go router, if there is another router with better performance.

File: internal/user/service_test.go (hunk 1 above)
Comment (dkischenko) [reply internal/user/service_test.go:689]: > 

For this case it's not important to use once. Just a copy-paste fail

**Review by Dimitriy14 — APPROVED** [review]

**Pull request merged**